api/net/Areas/Editor/Controllers/StorageController.cs

```diff
@@ -49,216 +55,627 @@ public StorageController(IOptions<StorageOptions> options)
+                    var capturePath = Path.Combine(locationPath, path);
+                    if (!client.Exists(capturePath)) throw new InvalidOperationException($"File does not exist: '{path}'");
+
+                    // Horrible implementation to download the whole capture file just to create a clip.
+                    var tmpPath = _connection.CopyFile(client, capturePath, _options.GetCapturePath(), this.User);
+                    var resultPath = await FfmpegHelper.CreateClipAsync(tmpPath, start, end, outputName);
+
+                    // Upload clip to remote location.
+                    using var clipStream = System.IO.File.OpenRead(resultPath);
+                    var clipPath = Path.Combine(capturePath.GetDirectoryPath(), Path.GetFileName(resultPath));
+                    client.UploadFile(clipStream, clipPath);
+                    clipStream.Close();
+
+                    var info = new ItemModel(Path.GetFileName(clipPath), client.GetAttributes(clipPath));
+
+                    // Delete temporary files.
+                    System.IO.File.Delete(tmpPath);
+                    System.IO.File.Delete(resultPath);
+
+                    return new JsonResult(info);
+                }
+                finally
+                {
+                    if (client.IsConnected)
+                        client.Disconnect();
+                }
+            }
+            else if (connection == null || connection?.ConnectionType == ConnectionType.LocalVolume)
+            {
+                var safePath = Path.Combine(_options.GetCapturePath(), path.MakeRelativePath());
+                var file = await FfmpegHelper.CreateClipAsync(safePath, start, end, outputName);
+                return new JsonResult(new ItemModel(file));
+            }
+            else throw new NotImplementedException($"Location connection type '{connection?.ConnectionType}' not implemented yet.");
+        }
+        else
+        {
+            var safePath = Path.Combine(_options.GetUploadPath(), path.MakeRelativePath());
+            var file = await FfmpegHelper.CreateClipAsync(safePath, start, end, outputName);
+            return new JsonResult(new ItemModel(file));
+        }
     }
 
     /// <summary>
     /// Make a clip from the target file identified in the clip parameter 'prefix'.
     /// </summary>
+    /// <param name="locationId"></param>
     /// <param name="path"></param>
     /// <param name="prefix"></param>
     /// <returns></returns>
     [HttpPost("join")]
+    [HttpPost("{locationId:int}/join")]
     [Produces(MediaTypeNames.Application.Json)]
     [ProducesResponseType(typeof(ItemModel), (int)HttpStatusCode.OK)]
     [ProducesResponseType((int)HttpStatusCode.BadRequest)]
     [SwaggerOperation(Tags = new[] { "Storage" })]
-    public async Task<IActionResult> JoinClipsAsync(string path, string prefix)
+    public async Task<IActionResult> JoinClipsAsync([FromRoute] int? locationId, [FromQuery] string path, [FromQuery] string prefix)
```

Comment:
I wish we could execute FFmpeg commands directly on the remote location, but it doesn't work.  Need to copy locally first.